### PR TITLE
SwitchCaseSpaceFixer - Fix spacing between 'case' and semicolon

### PR DESCRIPTION
**Pull request by SpacePossum**
@see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1656

### CODE REVIEW

File: Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function fix(\SplFileInfo $file, $content)
                     ++$ternariesCount;
                 }
 
-                if ($tokens[$colonIndex]->equals(':')) {
+                if ($tokens[$colonIndex]->equals(':') || $tokens[$colonIndex]->equals(';')) {
```
Comment (keradus) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: `equalsAny`

**Comment by keradus** [comment]
btw, 1.11.x

**Comment by keradus** [comment]
Please ignore fabbot.io this time

**Comment by gharlan** [comment]
> btw, 1.11.x

The fixer does not exist in 1.11 ;)

**Comment by keradus** [comment]
cool, thanks @gharlan !

File: Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php (hunk 1 above)
Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: check, done :)

File: Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ public function fix(\SplFileInfo $file, $content)
                     ++$ternariesCount;
                 }
 
-                if ($tokens[$colonIndex]->equals(':')) {
+                if ($tokens[$colonIndex]->equalsAny(array(':', ';'), false)) {
```
Comment (keradus) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: Plesae remove the last parameter, it wont be used in this case.

Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: the default is true, but it is not needed so false should be faster in execution doesn't it?

Comment (gharlan) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: No, a case-sensitive (`true`) check is faster than a case-insensitive (`false`).
But in this case we only have non-array tokens, so the parameter is not used:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/Tokenizer/Token.php#L117

Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/SwitchCaseSpaceFixer.php]: >Maybe you've misread the parameter as $caseInsensitive?

I did :|  thanks!

File: Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php [hunk 3]
```diff
@@ -272,6 +272,32 @@ public function provideFixCases()
...
+                '<?php
+                    switch($foo) {
+                        case 4  :  ; ;
+                        case 31 + test(";") ;  ; ; ;;
+                        case 1 + test(";") ; // ;
+                        case (1+2/*;*/) ;
+                        case 1  ;
+                        case 2 ;
+                            return 1;
+                        default;
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php]: would the space after `default` be removed?
(i bet yes, but nice to test;P)

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php]: add https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/2586e04d2ebb217643c94a1ba60e9915bbe8c142 :)

Comment (keradus) [inline Symfony/CS/Tests/Fixer/PSR2/SwitchCaseSpaceFixerTest.php]: thanks ;)

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**